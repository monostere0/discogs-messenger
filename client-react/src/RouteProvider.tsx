import * as React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link,
} from 'react-router-dom';

import List from './components/List';

export default function RouteProvider() {
  return (
    <Router>
      <div>
        <ul>
          <li><Link to="/">Foo</Link></li>
        </ul>

        <hr />

        <Route exact={true} path="/" component={List}/>
      </div>
    </Router>
  );
}