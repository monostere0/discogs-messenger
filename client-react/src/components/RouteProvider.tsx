import * as React from 'react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';

import MessengerContainer from './MessengerContainer';

export default function RouteProvider() {
  return (
    <Router>
      <MessengerContainer />
    </Router>
  );
}