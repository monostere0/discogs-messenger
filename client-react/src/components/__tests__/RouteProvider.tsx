import * as React from 'react';
import { create } from 'react-test-renderer';

import RouteProvider from '../RouteProvider';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: 'BrowserRouter',
}));
jest.mock('../MessengerContainer', () => ({
  default: 'MessengerContainer'
}));

describe('components/RouteProvider', () => {
  it('should render', () => {
    const tree = create(<RouteProvider />);
    expect(tree).toMatchSnapshot();
  });
});