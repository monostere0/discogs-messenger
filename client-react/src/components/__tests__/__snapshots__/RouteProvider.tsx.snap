// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RouteProvider should render 1`] = `
<BrowserRouter>
  <MessengerContainer />
</BrowserRouter>
`;
